### ISSUE TITLE
Datadog reports fails with 2.4.0

### ISSUE THREAD

**Issue by jvanbrunschot-coolblue** [DataDog/puppet-datadog-agent#488]
We updated to the latest version of this module (2.4.0) and the Puppet Datadog reporter fails. with the following error in puppetserver.log: 

`2019-01-18` 09:49:38,959 ERROR [qtp989190208-1139] [puppetserver] Puppet Report processor failed: Connection refused - Connection refused
org/jruby/ext/socket/RubyTCPSocket.java:126:in `initialize'
org/jruby/RubyIO.java:1197:in `open'
/opt/puppetlabs/server/apps/puppetserver/jruby-1_7.jar!/META-INF/jruby.home/lib/ruby/1.9/net/http.rb:763:in `connect'
org/jruby/ext/timeout/Timeout.java:115:in `timeout'
/opt/puppetlabs/server/apps/puppetserver/jruby-1_7.jar!/META-INF/jruby.home/lib/ruby/1.9/net/http.rb:763:in `connect'
/opt/puppetlabs/server/apps/puppetserver/jruby-1_7.jar!/META-INF/jruby.home/lib/ruby/1.9/net/http.rb:756:in `do_start'
/opt/puppetlabs/server/apps/puppetserver/jruby-1_7.jar!/META-INF/jruby.home/lib/ruby/1.9/net/http.rb:745:in `start'
/opt/puppetlabs/server/data/puppetserver/jruby-gems/gems/dogapi-1.32.0/lib/dogapi/common.rb:97:in `connect'
/opt/puppetlabs/server/data/puppetserver/jruby-gems/gems/dogapi-1.32.0/lib/dogapi/common.rb:117:in `request'
/opt/puppetlabs/server/data/puppetserver/jruby-gems/gems/dogapi-1.32.0/lib/dogapi/v1/metric.rb:24:in `upload'
/opt/puppetlabs/server/data/puppetserver/jruby-gems/gems/dogapi-1.32.0/lib/dogapi/v1/metric.rb:41:in `flush_buffer'
/opt/puppetlabs/server/data/puppetserver/jruby-gems/gems/dogapi-1.32.0/lib/dogapi/facade.rb:113:in `batch_metrics'
/etc/coolblue/puppet/modules/datadog_agent/lib/puppet/reports/datadog_reports.rb:115:in `process'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/report/processor.rb:37:in `process'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/report/processor.rb:53:in `processors'
org/jruby/RubyArray.java:1613:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/report/processor.rb:51:in `processors'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/report/processor.rb:30:in `process'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/report/processor.rb:14:in `save'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/indirector/indirection.rb:289:in `save'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/api/indirected_routes.rb:181:in `do_save'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/api/indirected_routes.rb:48:in `call'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/context.rb:65:in `override'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet.rb:260:in `override'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/api/indirected_routes.rb:47:in `call'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/route.rb:82:in `process'
org/jruby/RubyArray.java:1613:in `each'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/route.rb:81:in `process'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/route.rb:87:in `process'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/route.rb:87:in `process'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/handler.rb:64:in `process'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/profiler/around_profiler.rb:58:in `profile'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/util/profiler.rb:51:in `profile'
/opt/puppetlabs/puppet/lib/ruby/vendor_ruby/puppet/network/http/handler.rb:62:in `process'
file:/opt/puppetlabs/server/apps/puppetserver/puppet-server-release.jar!/puppetserver-lib/puppet/server/master.rb:42:in `handleRequest'
Puppet$$Server$$Master_1751441067.gen:13:in `handleRequest'
request_handler_core.clj:273:in `invoke'
jruby_request.clj:48:in `invoke'
jruby_request.clj:33:in `invoke'
request_handler_service.clj:47:in `handle_request'
request_handler.clj:3:in `invoke'
request_handler.clj:3:in `invoke'
core.clj:2515:in `invoke'
ring_middleware.clj:284:in `invoke'
core.clj:170:in `invoke'
core.clj:216:in `invoke'
core.clj:47:in `invoke'
core.clj:357:in `invoke'
core.clj:53:in `invoke'
ringutils.clj:83:in `invoke'
master_core.clj:742:in `invoke'
ring.cljc:25:in `invoke'
ring.cljc:16:in `invoke'
comidi.clj:245:in `invoke'
http.clj:152:in `invoke'
http.clj:152:in `invoke'
http.clj:148:in `invoke'
comidi.clj:332:in `invoke'
jetty9_core.clj:433:in `invoke'
normalized_uri_helpers.clj:74:in `invoke'`

puppetserver: 5.3.6-1.el7
puppet-agent: 5.5.8-1.el7
os: Amazon Linux 2

**Comment by truthbk** [comment]
Hi @jvanbrunschot-coolblue, it looks like a connectivity issue which may or may not be related to an issue with the module logic:
 
```
2019-01-18 09:49:38,959 ERROR [qtp989190208-1139] [puppetserver] Puppet Report processor failed: Connection refused - Connection refused
org/jruby/ext/socket/RubyTCPSocket.java:126:in initialize' org/jruby/RubyIO.java:1197:in open'
/opt/puppetlabs/server/apps/puppetserver/jruby-1_7.jar!/META-INF/jruby.home/lib/ruby/1.9/net/http.rb:763:in connect' org/jruby/ext/timeout/Timeout.java:115:in timeout'
```

could you please share your **redacted** `/etc/datadog-agent/datadog-reports.yaml` file (should be in your `master` node), that should help us get a sense of where we're trying to submit the report and hopefully find the root cause?

Thank you!

**Comment by cabrinha** [comment]
I'm having the same issue, getting connection refused: 
```2019-02-11 17:50:45,491 ERROR [qtp136368236-512] [puppetserver] Puppet Report processor failed: Connection refused - Connection refused```

Here is my /etc/datadog-agent/datadog-reports.yaml:
```### MANAGED BY PUPPET

---
:datadog_api_key: 'my_api_key_here'
:api_url: api.datadoghq.com
```

**Comment by truthbk** [comment]
Should be fixed in the merged PRs and upcoming release `2.5.0`.
